app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java

```diff
@@ -93,6 +95,10 @@ public void onSlide(@NonNull View bottomSheet, float slideOffset) {
         preferencesManager.increaseNumAppOpens();
         if (preferencesManager.getNumAppOpens() == NUM_APP_OPENS_FOR_RATING_ASK) {
             showPleaseRateDialog();
+        }else if (preferencesManager.getNumAppOpens() % NUM_APP_OPENS_FOR_RESTORE_AND_BACKUP == 0 &&
+                !preferencesManager.hasSeenBackupAndRestore()){
```

Comment:
We should also check against whether the feature is already in use. I think we persist the backup file path to shared preferences, so we can check against that being non-null and non-empty.
```suggestion
                !preferencesManager.hasSeenBackupAndRestore()) {
```